webservice/template/admin/logs/index.tmpl

```diff
@@ -0,0 +1,191 @@
+{{define "title"}}Logs{{end}}
+{{define "head"}}
+    <link rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.14.2/dist/bootstrap-table.min.css">
+    <style>
+        .id-col {
+            font-weight: 700;
+        }
+    </style>
+{{end}}
+{{define "content"}}
+    <h2 class="display-4 mb-3">Logs</h2>
+    <hr>
+
+    <div class="row">
+        <div class="col">
+            {{ if eq ( len .Logs ) 0}}
+                <h4 class="display-6 mb-3">No logs available</h4>
+            {{ else }}
+
+                <table class="table table-hover table-striped table-sm" id="table" data-toggle="table"
+                       data-pagination="true" data-search="true" data-page-size="100">
+                    <thead class="thead-light">
+                    <tr>
+                        <th data-sortable="true" scope="col">#</th>
+                        <th data-sortable="true" scope="col">Name</th>
+                        <th data-sortable="true" scope="col">Activity ID</th>
+                        <th data-sortable="true" scope="col">Explanation</th>
+                        <th data-sortable="true" scope="col">Role</th>
+                        <th data-sortable="true" scope="col">Type</th>
+                        <th data-sortable="true" scope="col">Date</th>
+                    </tr>
+                    </thead>
+                    <tbody>
+                    {{range .Logs}}
+
+                        <!-- Check if user is not deleted -->
+                        <tr {{if or (not .UserName.Valid) (or (lt .Activity 0) (gt .Activity 124) )}}class="table-danger"{{end}}>
+                            <td class="id-col">{{ .UserID }}</td>
+                            <td>{{if .UserName.Valid}}
+                                    {{ .UserName.Value }}
+                                {{else}}
+                                    N/A
+                                {{end}}
+                            </td>
+                            <td>{{ .Activity }}</td>
+                            <td>
+
+                                <!-- Declare new variable for affected user name -->
+                                {{ $U := ""}}
+
+                                <!-- Convert from int64 to int and get name if it's valid-->
+                                {{ if .AffectedUserID.Valid}}
+                                    {{ $A := INT64_TO_INT .AffectedUserID.Value}}
+                                    {{ $U = GET_USERNAME $A}}
+
+                                    <!-- If user doesn't exist, create a response string -->
+                                    {{ if eq $U ""}}
+                                        {{ $U = APPEND "user with ID " (ITOA $A)}} <!-- Convert int to string also -->
+                                    {{end}}
+                                {{end}}
+
+                                <!-- USER events -->
+                                {{ if eq .Activity 0 }} <!-- User activity:  New user-->
```

Comment:
This could be a plugin.
Easier to edit maybe? And clean looking in the template